### PR TITLE
exclude top k from max vio calculation to try and skip the "padding experts"

### PR DESCRIPTION
**Pull request by Jackmin801**
> [!NOTE]
> Updates `get_load_balance_stats` to optionally ignore top-k routed experts (avoiding padding experts) when computing max violation, with proper buffer reset.
> 
> - **MoE Load Balance Stats (`src/prime_rl/trainer/model.py`)**:
>   - Add `try_to_avoid_padding_experts: bool = True` to `get_load_balance_stats`.
>   - When enabled, sort `mlp.tokens_per_expert` desc and slice off `router.top_k` to exclude top-k routed experts from mean/max calc.
>   - Ensure reset zeros the original `mlp.tokens_per_expert` buffer.
>   - Add explicit tensor type annotation for `tokens_per_expert`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a1d15ea339d19e448be2f45cec017b5527219e99. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/trainer/model.py [hunk 1]
```diff
@@ -51,15 +51,20 @@ def is_tt_moe_model(model: nn.Module) -> bool:
...
     per_layer_max_vio = []
     for transformer_block in model.model.layers:
         # This is necessary for models that have mixed dense layers
         if not hasattr(transformer_block.mlp, "tokens_per_expert"):
             continue
         tokens_per_expert = transformer_block.mlp.tokens_per_expert
         balanced_load = tokens_per_expert.mean()
-        max_vio = (tokens_per_expert.max() - balanced_load) / balanced_load
+        if use_second_popular_expert:
+            max_vio = (tokens_per_expert.topk(2).values[-1] - balanced_load) / balanced_load
```
[the remark this answers is not in the document]
Comment (Jackmin801) [reply src/prime_rl/trainer/model.py]: i dont think this is an issue because I think each expert parallel shard should store the full tokens_per_expert though I could be wrong.

[the remark this answers is not in the document]
Comment (Jackmin801) [reply src/prime_rl/trainer/model.py]: the issue would also make the metric wrong in the max case anyways

**Review by mikasenghaas — COMMENTED** [review]
this would make us blind to very heavy expert imbalance where all tokens are routed to a singular expert?

**Comment by Jackmin801** [comment]
yea it would but also its better than us not caring about the metric because we always think its just padding tokens ruining the metric

**Review by mikasenghaas — APPROVED** [review]
